### DISCUSSION TITLE
Generation of a wheel-package for specific python version using pyproject.toml

### DISCUSSION THREAD

**Discussion by anates** [python-poetry/poetry#9576] [General]
I'm currently transferring my project build pipeline from being` setup.py`-based to a `pyproject.toml`-based approach. Most of the parameters are easily transferable, however, I can't find an equivalent to 

> python_requires="~=3.11"

Of course, there is

> [project]
> requires-python = "~=3.11"

but while the setting in `setup.py` results in 

> -cp311-cp311

 in the package name, the same setting in `pyproject.toml` results in 

> -py3-none

 in the package name.

As the package contains several dll-files which have to be compiled specific to each python version, I want to reflect that also in the package name (and ideally refuse installation if a wrong python installation is used). How can I solve that by using `pyproject.toml`? Similarly, how can I build packages specifically for Windows/Unix/MacOS using this approach, something which is possible using `setup.py`?

**Comment by dimbleby** [comment]
#2740
